nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/message/BulkCommand.java

```diff
@@ -159,7 +158,31 @@ public void setRepository(String repository) {
          *
          * @param action the registered bulk action name
          * @param nxqlQuery the query that represent the document set to apply the action
+         * @param user the user with whose rights the computation will be executed
          */
+        public Builder(String action, String nxqlQuery, String username) {
+            if (isEmpty(action)) {
+                throw new IllegalArgumentException("Action cannot be empty");
+            }
+            this.action = action;
+            if (isEmpty(nxqlQuery)) {
+                throw new IllegalArgumentException("Query cannot be empty");
+            }
+            this.query = nxqlQuery;
+            if (isEmpty(username)) {
+                throw new IllegalArgumentException("Username cannot be empty");
+            }
+            this.username = username;
+        }
+
+        /**
+         * BulkCommand builder
+         *
+         * @param action the registered bulk action name
+         * @param nxqlQuery the query that represent the document set to apply the action
+         * @deprecated 11.1 use constructor with username instead
```

Comment:
`@deprecated 11.1 use constructor with username instead` -> `@deprecated since 11.1`, use {@link #Builder(String, String, String)} constructor with username instead.